src/chariot/database/Database.py

```diff
@@ -0,0 +1,60 @@
+import abc
+from typing import List
+
+
+class Database(metaclass=abc.ABCMeta):
+    def __init__(self, db_path: str = '../data/iot_database.db'):
+        self.db_path = db_path
+
+    @abc.abstractmethod
+    def initializeDB(self):
+        """
+        Create the necessary table in the database
+        """
+        pass
+
+    @abc.abstractmethod
+    def connectToDB(self):
+        """
+        Establish a database connection
+        """
+        pass
+
+    @abc.abstractmethod
+    def deleteDB(self):
+        """
+        Delete the database file
+        """
+        pass
+
+    @abc.abstractmethod
+    def disconnectFromDB(self):
+        """
+        Delete the database connection
+        """
+        pass
+
+    @abc.abstractmethod
+    def insertOne(self, dataPoint: dict):
+        """
+        Check for validity of dataPoint, then insert into database
+        """
+        if sorted(dataPoint.keys()) != sorted(['relative_time', 'freeform']):
```

Comment:
This seems like an awkward way to check for these. The way this is done, one cannot tell which field was invalid or missing. You can define the `set` of required fields as a class variable and just iterate through the keys to see if
1. Every key in the dict is in the required fields
2. No other key apart from the required ones exist in the dict

Iterating will allow you to know exactly where this dict was flagged as invalid, and you can throw an error accordingly.